

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Demos &mdash; EpiMapper  documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=e59714d7" />

  
      <script src="../_static/jquery.js?v=5d32c60e"></script>
      <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../_static/documentation_options.js?v=5929fcd5"></script>
      <script src="../_static/doctools.js?v=9a2dae69"></script>
      <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Required Files and Where to Find Them" href="Required_files_and_where_to_find_them.html" />
    <link rel="prev" title="differential_analysis" href="differential_analysis.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: white" >

          
          
          <a href="../index.html" class="icon icon-home">
            EpiMapper
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="fastqc.html">fastqc</a></li>
<li class="toctree-l1"><a class="reference internal" href="bowtie2_alignment.html">bowtie2_alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="remove_duplicates.html">remove_duplicates</a></li>
<li class="toctree-l1"><a class="reference internal" href="fragment_length.html">fragment_length</a></li>
<li class="toctree-l1"><a class="reference internal" href="filtering.html">filtering</a></li>
<li class="toctree-l1"><a class="reference internal" href="spike_in_calibration.html">spike_in_calibration</a></li>
<li class="toctree-l1"><a class="reference internal" href="peak_calling.html">peak_calling</a></li>
<li class="toctree-l1"><a class="reference internal" href="heatmap.html">heatmap</a></li>
<li class="toctree-l1"><a class="reference internal" href="differential_analysis.html">differential_analysis</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Demos</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#cut-tag-human-histone-modification-demo">CUT&amp;Tag Human Histone Modification Demo</a></li>
<li class="toctree-l2"><a class="reference internal" href="#atac-seq-demo">ATAC-seq Demo</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Required_files_and_where_to_find_them.html">Required Files and Where to Find Them</a></li>
<li class="toctree-l1"><a class="reference internal" href="further_analysis_example.html">Further analysis - Example</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu"  style="background: white" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">EpiMapper</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Demos</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/Demos.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="demos">
<h1>Demos<a class="headerlink" href="#demos" title="Link to this heading"></a></h1>
<p>There are currently two demos avalible for using the EpiMapper Python package, one for CUT&amp;Tag data and one for ATAC-seq data.
Arguments</p>
<section id="cut-tag-human-histone-modification-demo">
<h2>CUT&amp;Tag Human Histone Modification Demo<a class="headerlink" href="#cut-tag-human-histone-modification-demo" title="Link to this heading"></a></h2>
<p>The data used in this demo is collected from GEO association: GSE145187 derived from Kaya-Okur et al. (2020).</p>
<p>This dataset consists of 6 samples, targeting two histone modifications: H3K4me3 and H3K27me3 with two replicates each, as well as two IgG control samples.</p>
<p>You need to download and create two folders:</p>
<ul class="simple">
<li><p>fastq - containing demo FASTQ files from CUT&amp;Tag, downloaded from : <a class="reference external" href="https://zenodo.org/records/10822274">https://zenodo.org/records/10822274</a></p></li>
<li><p>in - containing all necessary input files for EpiMapper usage, downloaded from <a class="reference external" href="https://zenodo.org/records/10822349">https://zenodo.org/records/10822349</a></p></li>
</ul>
<p>Additionally, you need to create an “out” folder where the output will be stored. Folders can be created by using:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>mkdir<span class="w"> </span>out
</pre></div>
</div>
<p>The script to run the demo is shown below:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># 1. fastqc</span>

epimapper<span class="w"> </span>fastqc<span class="w"> </span>-f<span class="w"> </span>fastq<span class="w"> </span>-o<span class="w"> </span>out

<span class="c1"># 2. bowtie2_alignment - To reference genome Hg38</span>

epimapper<span class="w"> </span>bowtie2_alignment<span class="w"> </span>-f<span class="w"> </span>fastq<span class="w"> </span>-i<span class="w"> </span><span class="k">in</span>/bowtie2_index_hg38<span class="w">  </span>-m<span class="w"> </span>True<span class="w"> </span>-o<span class="w"> </span>out

<span class="c1"># bowtie2_alignment - To spike-in genome (E.coli)</span>

epimapper<span class="w"> </span>bowtie2_alignment<span class="w"> </span>-f<span class="w"> </span>fastq<span class="w"> </span>-s<span class="w"> </span>True<span class="w"> </span>-i<span class="w"> </span><span class="k">in</span>/bowtie2_index_ecoli<span class="w"> </span>-m<span class="w"> </span>True<span class="w"> </span>-o<span class="w"> </span>out
</pre></div>
</div>
<p><strong>Example table</strong>: After running <code class="docutils literal notranslate"><span class="pre">bowtie2_alignment</span></code>, the result (Table 1) appears as follows:</p>
<table class="my-custom-class docutils align-center" id="id1">
<caption><span class="caption-text">Table 1: Example bowtie2_alignment_ref_and_spike_in</span><a class="headerlink" href="#id1" title="Link to this table"></a></caption>
<thead>
<tr class="row-odd"><th class="head"><p>Sample</p></th>
<th class="head"><p>Replication</p></th>
<th class="head"><p>SequencingDepth</p></th>
<th class="head"><p>MappedFragments</p></th>
<th class="head"><p>AlignmentRate</p></th>
<th class="head"><p>MappedFragments_SpikeIn</p></th>
<th class="head"><p>AlignmentRate_SpikeIn</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>d-H3K27me3</p></td>
<td><p>rep1</p></td>
<td><p>729951</p></td>
<td><p>729478</p></td>
<td><p>99.94%</p></td>
<td><p>473</p></td>
<td><p>0.06%</p></td>
</tr>
<tr class="row-odd"><td><p>d-H3K27me3</p></td>
<td><p>rep2</p></td>
<td><p>695765</p></td>
<td><p>695534</p></td>
<td><p>99.97%</p></td>
<td><p>231</p></td>
<td><p>0.03%</p></td>
</tr>
<tr class="row-even"><td><p>d-H3K4me3</p></td>
<td><p>rep1</p></td>
<td><p>358119</p></td>
<td><p>357764</p></td>
<td><p>99.9%</p></td>
<td><p>355</p></td>
<td><p>0.1%</p></td>
</tr>
<tr class="row-odd"><td><p>d-H3K4me3</p></td>
<td><p>rep2</p></td>
<td><p>472641</p></td>
<td><p>468294</p></td>
<td><p>99.08%</p></td>
<td><p>4347</p></td>
<td><p>0.92%</p></td>
</tr>
<tr class="row-even"><td><p>d-IgG</p></td>
<td><p>rep1</p></td>
<td><p>134669</p></td>
<td><p>59177</p></td>
<td><p>43.94%</p></td>
<td><p>75492</p></td>
<td><p>56.06%</p></td>
</tr>
<tr class="row-odd"><td><p>d-IgG</p></td>
<td><p>rep2</p></td>
<td><p>603373</p></td>
<td><p>482561</p></td>
<td><p>79.98%</p></td>
<td><p>120812</p></td>
<td><p>20.02%</p></td>
</tr>
</tbody>
</table>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># 3. remove_duplicates</span>

epimapper<span class="w"> </span>remove_duplicates<span class="w"> </span>-s<span class="w"> </span>out/Epimapper/alignment/sam<span class="w"> </span>-o<span class="w"> </span>out

<span class="c1"># 4. fragment_length</span>

epimapper<span class="w"> </span>fragment_length<span class="w"> </span>-s<span class="w"> </span>out/Epimapper/alignment/removeDuplicate/sam_duplicates_removed<span class="w"> </span>-o<span class="w"> </span>out
</pre></div>
</div>
<p><strong>Example Plot</strong>: After running <code class="docutils literal notranslate"><span class="pre">fragment_length</span></code>, the fragment length distribution plot (Figure 1 and Figure 2) appears as follows:</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/fragment_histone.png"><img alt="ViolinPlot Fragment Length" src="../_images/fragment_histone.png" style="width: 100%;" />
</a>
</figure>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># 5. filtering</span>

epimapper<span class="w"> </span>filtering<span class="w"> </span>-s<span class="w"> </span>out/Epimapper/alignment/removeDuplicate/sam_duplicates_removed<span class="w"> </span><span class="se">\</span>
-cs<span class="w"> </span><span class="k">in</span>/hg38.chrom.sizes.clear.sorted<span class="w"> </span>-bl<span class="w"> </span><span class="k">in</span>/blacklist.bed<span class="w">  </span>-sn<span class="w"> </span>True<span class="w"> </span>-o<span class="w"> </span>out

<span class="c1"># 6. spike_in_calibration</span>

epimapper<span class="w"> </span>spike_in_calibration<span class="w"> </span>-b<span class="w"> </span>out/Epimapper/alignment/bed<span class="w"> </span>-cs<span class="w"> </span><span class="k">in</span>/hg38.chrom.sizes.clear.sorted<span class="w"> </span><span class="se">\</span>
-ss<span class="w"> </span>out/Epimapper/alignment/sam_spike_in<span class="w"> </span>-o<span class="w"> </span>out

<span class="c1"># 7. peak_calling</span>

epimapper<span class="w"> </span>peak_calling<span class="w">  </span>-soft<span class="w"> </span>seacr<span class="w"> </span>-f<span class="w"> </span>out/Epimapper/alignment/bed<span class="w"> </span>-bg<span class="w"> </span>out/Epimapper/alignment/bedgraph<span class="w"> </span><span class="se">\</span>
-c<span class="w"> </span>IgG<span class="w"> </span>-o<span class="w"> </span>out

<span class="c1"># 8. heatmap</span>

epimapper<span class="w"> </span>heatmap<span class="w"> </span>-b<span class="w"> </span>out/Epimapper/alignment/bam<span class="w">  </span>-p<span class="w"> </span>out/Epimapper/peakCalling/seacr/control<span class="w"> </span><span class="se">\</span>
-bl<span class="w"> </span><span class="k">in</span>/blacklist.bed<span class="w"> </span>-r<span class="w"> </span><span class="k">in</span>/hg38.refFlat.txt<span class="w">  </span>-o<span class="w"> </span>out
</pre></div>
</div>
<p><strong>Example Plot</strong>: After running <code class="docutils literal notranslate"><span class="pre">heatmap</span></code>, the heatmap of histone enrichment around genes (Figure 3 and Figure 4) appears as follows. Figure 4 is a composite heatmap constructed from individual peak heatmaps of single samples, represented as a single image in the original file.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/heatmap_histone.png"><img alt="Heatmap" src="../_images/heatmap_histone.png" style="width: 100%;" />
</a>
</figure>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># 9. differential_analysis</span>

epimapper<span class="w"> </span>differential_analysis<span class="w"> </span>-p<span class="w"> </span>out/Epimapper/peakCalling/seacr/control<span class="w"> </span><span class="se">\</span>
-bg<span class="w"> </span>out/Epimapper/alignment/bedgraph<span class="w"> </span><span class="se">\</span>
-bl<span class="w"> </span><span class="k">in</span>/blacklist.bed<span class="w"> </span>-r<span class="w"> </span><span class="k">in</span>/hg38.refFlat.txt<span class="w"> </span>-cs<span class="w"> </span><span class="k">in</span>/hg38.chrom.sizes.clear.sorted<span class="w"> </span><span class="se">\</span>
-la<span class="w"> </span>H3K27me3_rep1<span class="w"> </span>H3K27me3_rep2<span class="w"> </span>-lb<span class="w"> </span>H3K4me3_rep1<span class="w"> </span>H3K4me3_rep2<span class="w"> </span>-an<span class="w"> </span>True<span class="w"> </span><span class="se">\</span>
-e<span class="w">  </span><span class="k">in</span>/hg38_all_enhancers_merged_hglft_genome_327b3_4dmr.bed<span class="w"> </span>-o<span class="w"> </span>out
</pre></div>
</div>
</section>
<section id="atac-seq-demo">
<h2>ATAC-seq Demo<a class="headerlink" href="#atac-seq-demo" title="Link to this heading"></a></h2>
<p>The data used in this demo is from an ATAC-seq experiment of healthy/diabetic pancreatic islet, collected from Brysani et al (2020) with the GEO assositation: GSE129383.</p>
<p>Here, the demo data only contians the chr21 from the orginal data, to save space. This dataset conists of ATAC-seq data from 6 diabetic donors and 9 healthy donors, only one replicate from each sample. The data avalible for this demo is publicly avalible at a zenodo:</p>
<p>You need to download and create two folders:</p>
<ul class="simple">
<li><p>fastq - containing demo FASTQ files from ATAC-seq, downloaded from : <a class="reference external" href="https://zenodo.org/records/10818453">https://zenodo.org/records/10818453</a></p></li>
<li><p>in - containing all necessary input files for EpiMapper usage, downloaded from <a class="reference external" href="https://zenodo.org/records/10818469">https://zenodo.org/records/10818469</a></p></li>
</ul>
<p>Additionally, you need to create an “out” folder where the output will be stored. Folders can be created by using:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>mkdir<span class="w"> </span>out
</pre></div>
</div>
<p>The script to run the demo is shown below:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># EpiMapper demo run on human ATAC-seq data (only chr21)</span>

<span class="c1"># 1. fastqc</span>

epimapper<span class="w"> </span>fastqc<span class="w"> </span>-f<span class="w"> </span>fastq<span class="w"> </span>-o<span class="w"> </span>out

<span class="c1"># 2. bowtie2_alignment</span>

epimapper<span class="w"> </span>bowtie2_alignment<span class="w"> </span>-f<span class="w"> </span>fastq<span class="w"> </span>-i<span class="w"> </span><span class="k">in</span>/hg19_chr21_bowtie2_index<span class="w"> </span>-o<span class="w"> </span>out

<span class="c1"># 3. remove_duplicates</span>

epimapper<span class="w"> </span>remove_duplicates<span class="w"> </span>-s<span class="w"> </span>out/Epimapper/alignment/sam<span class="w"> </span>-o<span class="w"> </span>out

<span class="c1"># 4. fragment_length</span>

epimapper<span class="w"> </span>fragment_length<span class="w"> </span>-s<span class="w"> </span>out/Epimapper/alignment/removeDuplicate/sam_duplicates_removed<span class="w"> </span>-o<span class="w"> </span>out

<span class="c1"># 5. filtering</span>

epimapper<span class="w"> </span>filtering<span class="w"> </span>-s<span class="w"> </span>/Users/eier/Documents/demo/ATAC/out/Epimapper/alignment/removeDuplicate/sam_duplicates_removed<span class="w"> </span><span class="se">\</span>
-cs<span class="w"> </span><span class="k">in</span>/hg19_chromosome_sizes_sorted.txt<span class="w"> </span>-bl<span class="w"> </span><span class="k">in</span>/hg19-blacklist_sorted.bed<span class="w"> </span>-atac<span class="w"> </span>True<span class="w"> </span>-o<span class="w"> </span>/Users/eier/Documents/demo/ATAC/out

<span class="c1"># 6. peak_calling</span>

epimapper<span class="w"> </span>peak_calling<span class="w"> </span>-soft<span class="w"> </span>macs2<span class="w"> </span>-f<span class="w"> </span>/Users/eier/Documents/demo/ATAC/out/Epimapper/alignment/bed<span class="w"> </span>-b<span class="w"> </span>/Users/eier/Documents/demo/ATAC/out/Epimapper/alignment/bam<span class="w"> </span><span class="se">\</span>
-gs<span class="w"> </span><span class="m">2</span>.7e9<span class="w">  </span>-o<span class="w"> </span>/Users/eier/Documents/demo/ATAC/out

<span class="c1"># 7. heatmaps</span>

epimapper<span class="w"> </span>heatmap<span class="w"> </span>-b<span class="w"> </span>out/Epimapper/alignment/bam<span class="w"> </span>-bl<span class="w"> </span><span class="k">in</span>/hg19-blacklist_sorted.bed<span class="w"> </span><span class="se">\</span>
-p<span class="w"> </span>out/Epimapper/peakCalling/macs2/top_peaks<span class="w"> </span>-r<span class="w"> </span><span class="k">in</span>/hg19.refFlat_chr21.txt<span class="w"> </span>-o<span class="w"> </span>/Users/eier/Documents/demo/ATAC/out


<span class="c1">#8. differntial_analysis</span>

epimapper<span class="w"> </span>differential_analysis<span class="w"> </span>-p<span class="w"> </span>out/Epimapper/peakCalling/macs2/top_peaks<span class="w"> </span><span class="se">\</span>
-r<span class="w"> </span><span class="k">in</span>/hg19.refFlat_chr21.txt<span class="w">  </span>-bl<span class="w"> </span><span class="k">in</span>/hg19-blacklist_sorted.bed<span class="w"> </span>-cs<span class="w"> </span><span class="k">in</span>/hg19_chromosome_sizes_sorted_filtered.txt<span class="w"> </span><span class="se">\</span>
-fold<span class="w"> </span>True<span class="w"> </span>-an<span class="w"> </span>True<span class="w"> </span>-e<span class="w"> </span><span class="k">in</span>/hg19_all_enhancers_merged_4dmr.bed<span class="w"> </span>-o<span class="w"> </span>out<span class="w"> </span><span class="se">\</span>
-la<span class="w"> </span>diabetic-1_rep1<span class="w"> </span>diabetic-2_rep1<span class="w"> </span>diabetic-3_rep1<span class="w"> </span>diabetic-4_rep1<span class="w"> </span>diabetic-5_rep1<span class="w"> </span>diabetic-6_rep1<span class="w"> </span><span class="se">\</span>
-lb<span class="w"> </span>healthy-1_rep1<span class="w"> </span>healthy-2_rep1<span class="w"> </span>healthy-3_rep1<span class="w"> </span>healthy-4_rep1<span class="w"> </span>healthy-5_rep1<span class="w"> </span>healthy-6_rep1<span class="w"> </span>healthy-7_rep1<span class="w"> </span>healthy-8_rep1<span class="w"> </span>healthy-9_rep1
</pre></div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="differential_analysis.html" class="btn btn-neutral float-left" title="differential_analysis" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="Required_files_and_where_to_find_them.html" class="btn btn-neutral float-right" title="Required Files and Where to Find Them" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Jenny Sofie Dragland &amp; Junbai Wang.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>